src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -314,7 +314,7 @@ public async Task DeleteSubscription(string name, string raftRequestId)
 
         public bool DropSubscriptionConnections(long subscriptionId, SubscriptionException ex)
         {
-            if (_subscriptions.TryRemove(subscriptionId, out SubscriptionConnectionsState subscriptionConnectionsState) == false)
+            if (_subscriptions.TryGetValue(subscriptionId, out SubscriptionConnectionsState subscriptionConnectionsState) == false)
```

Comment:
Doesn't this mean that we are going to hang around in the `_subscriptions` for a while? If we have a lot of connections, and we drop them all, what will clean it up?

Reply:
`_subscriptions` hold the state for each subscription, those are cleaned on:

- `IdleOperations` run 
- LowMemory

Each state holds `_connections` which are _fully_ cleaned on:

- `/databases/*/subscriptions/drop`  request
- On `SubscriptionChangeVectorUpdateConcurrencyException`  (subscription CV was changed during execution)
- Database Deleted
- Subscription Task disabled
- Subscription Task responsible node changed
- Subscription Query was changed
- State disposse

Reply:
Okay, good enough for me.